File: Day319/Largest3SameDigitNubmerInString.java
class Solution {
    public String largestGoodInteger(String num) {
        if (num.length() <=2) {
            return "";
        }
        int max = -1;
        for (int i=2; i<num.length(); i++) {
            if (num.charAt(i-2) == num.charAt(i-1) && num.charAt(i-1) == num.charAt(i)) {
                if ((num.charAt(i)-'0')>max) {
                    max = num.charAt(i)-'0';
                }
            }
        }
        return max == -1? "": (""+max+""+max+""+max);
    }
}

File: Day319/Largest3SameDigitNubmerInString.go
func largestGoodInteger(num string) string {
    arr := make([]int, 10)
    for i:=0; i<len(num); i++ {
        if i+2 < len(num) {
            if num[i] == num[i+1] && num[i] == num[i+2] {
                arr[num[i]-'0']++
            }
        }
    }
    for i:=9; i>=0; i-- {
        if arr[i] != 0 {
            a := string(i+'0')
            return a+a+a
        }
    }
    return "";
}